src/Config.zig

```diff
@@ -202,10 +202,13 @@ pub fn configChanged(config: *Config, allocator: std.mem.Allocator, builtin_crea
         config.builtin_path = try std.fs.path.join(allocator, &.{ builtin_creation_dir.?, "builtin.zig" });
     }
 
+    const cache_dir_path = (try known_folders.getPath(allocator, .cache)) orelse {
```

Comment:
You should append `zls` to this. It's bad practice to drop stuff straight in `~/.cache`.